### PR TITLE
[C-API] Bugfix for abnormal accessing of the callback function

### PR DESCRIPTION
**Pull request by again4you**
If the handle of sink-type element (i.e. tensor_sink and app_sink) is fetched by calling both `ml_pipeline_element_get_handle()` and `ml_pipeline_sink_register()` functions, segmantation fault occurs because of abnormal accessing for callback function. This PR adds this case and fixes this bug

It contains the following patches.
* Bugfix for abnormal accessing of the callback function
* Add test case for abnormal accessing of callback function

Signed-off-by: Sangjung Woo <sangjung.woo@samsung.com>

**Self evaluation:**
1. Build test: [*]Passed [ ]Failed []Skipped
2. Run test: [*]Passed [ ]Failed []Skipped

#### Submit v2
* Use `g_list_free_full()` and `GDestroyNotify` instead of iteration of GList.
#### Submit v1
* First draft

### CODE REVIEW

File: api/capi/src/nnstreamer-capi-pipeline.c [hunk 1]
```diff
@@ -332,8 +337,17 @@ cleanup_node (gpointer data)
     e->handle_id = 0;
   }
 
-  if (e->handles)
-    g_list_free_full (e->handles, g_free);
+  l = e->handles;
```
Comment (jaeyun-jung) [inline api/capi/src/nnstreamer-capi-pipeline.c:340]: simply how about adding cb free function?
```
static void
free_element_handle (gpointer data)
{
  ml_pipeline_common_elem *item = (ml_pipeline_common_elem *) data;
  g_free (item->callback_info);
  g_free (item);
}

if (e->handles)
  g_list_free_full (e->handles, free_element_handle);
```

Comment (again4you) [reply api/capi/src/nnstreamer-capi-pipeline.c:340]: Nice catch! I will

File: tests/tizen_capi/unittest_tizen_capi.cc [hunk 2]
```diff
@@ -5949,6 +5949,52 @@ TEST (nnstreamer_capi_element, get_property_04_n)
   EXPECT_EQ (status, ML_ERROR_NONE);
 }
 
+TEST (nnstreamer_capi_element, scenario_05_p)
```
Comment (myungjoo) [inline tests/tizen_capi/unittest_tizen_capi.cc:5952]: Add negative cases please!

**Comment by myungjoo** [comment]
We can't proceed unless you add at least two negative cases. The negative-case ratio is under the required threshold.

**Pull request merged**